From d7b8dcbecdc40d98d20812176f57d7f72fa3f6b9 Mon Sep 17 00:00:00 2001
From: liubo <liubo254@huawei.com>
Date: Fri, 16 Jul 2021 10:09:02 +0800
Subject: [PATCH] move ETMEM feature CONFIG to mm/Kconfig and add architecture
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: d7b8dcbecdc40d98d20812176f57d7f72fa3f6b9

 dependency

euleros inclusion
category: feature
feature: etmem
bugzilla: 48246

-------------------------------------------------

The original etmem feature failed to compile a specific architecture,
for example, powerpc, because the architecture is not specified.

This patch is move ETMEM feature CONFIG to mm/Kconfig and add
architecture.

Signed-off-by: liubo <liubo254@huawei.com>
Reviewed-by: jingxiangfeng 00447129 <jingxiangfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 lib/Kconfig | 11 -----------
 mm/Kconfig  | 15 +++++++++++++++
 2 files changed, 15 insertions(+), 11 deletions(-)

diff --git a/lib/Kconfig b/lib/Kconfig
index cb5e620bf100..b46a9fd122c8 100644
--- a/lib/Kconfig
+++ b/lib/Kconfig
@@ -660,17 +660,6 @@ config PARMAN
 config OBJAGG
 	tristate "objagg" if COMPILE_TEST
 
-config ETMEM_SCAN
-	tristate "module: etmem page scan for etmem support"
-	help
-	 etmem page scan feature
-	 used to scan the virtual address of the target process
-
-config ETMEM_SWAP
-       tristate "module: etmem page swap for etmem support"
-       help
-         etmem page swap feature
-
 config STRING_SELFTEST
 	tristate "Test string functions"
 
diff --git a/mm/Kconfig b/mm/Kconfig
index e76ecfae6d45..398ddb19155e 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -498,6 +498,21 @@ config MEMCG_QOS
 
 	  If unsure, say "n".
 
+config ETMEM_SCAN
+	tristate "module: etmem page scan for etmem support"
+        depends on MMU
+        depends on X86 || ARM64
+	help
+	 etmem page scan feature
+	 used to scan the virtual address of the target process
+
+config ETMEM_SWAP
+	tristate "module: etmem page swap for etmem support"
+	depends on MMU
+	depends on X86 || ARM64
+	help
+	 etmem page swap feature
+
 config CMA
 	bool "Contiguous Memory Allocator"
 	depends on MMU
-- 
2.26.2

